modules/core/src/main/scala/org/scalasteward/core/edit/update/data/ModulePosition.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018-2022 Scala Steward contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.scalasteward.core.edit.update.data
+
+sealed trait ModulePosition extends Product with Serializable {
+  def groupId: Substring.Position
+  def artifactId: Substring.Position
+  def version: Substring.Position
+
+  final def unwrappedVersion: String =
+    version.value.dropWhile(Set('"', '$', '{')).reverse.dropWhile(Set('"', '}')).reverse
+}
+
+object ModulePosition {
+  final case class SbtDependency(
+      groupId: Substring.Position,
+      artifactId: Substring.Position,
+      version: Substring.Position
+  ) extends ModulePosition
+
+  final case class MillDependency(
+      groupId: Substring.Position,
+      artifactId: Substring.Position,
+      version: Substring.Position
+  ) extends ModulePosition
+
+  final case class MavenDependency(
+      groupId: Substring.Position,
+      artifactId: Substring.Position,
+      version: Substring.Position
+  ) extends ModulePosition
```

Comment:
**question:** why do we need to differentiate between these three? are we expecting projects to use more than one build system?

<sub>This comment follows the [**conventionalcomments.org**](https://conventionalcomments.org) standard</sub>

Reply:
We currently don't need these three. `ModulePosition` could just be a case class with the three `Position` fields. I guess I added these because `ModuleScanner` has different `find` methods for each of them. But we currently don't differentiate between the types of `ModulePosition` later.

I'll change them to a single case class.

Reply:
Done in [30e1cd4](https://github.com/scala-steward-org/scala-steward/pull/2816/commits/30e1cd4c1c556b7ce23af535a18727bea846938f).

Reply:
> are we expecting projects to use more than one build system?

Yes, projects that use more than one build system exist. One example is https://github.com/com-lihaoyi/scalatags which has a sbt and Mill build. And Scala Steward uses both to extract dependencies.

Reply:
Oh! Great to know that 😊 thanks!